### DISCUSSION TITLE
How to map the key `ç` at `hiragana-romanji.tsv`?

### DISCUSSION THREAD

**Discussion by paulolimac** [google/mozc#479] [Q&A]
Without success, I can't map the key `ç` with ibus-mozc.
But with success, I can map `b'` to be converted in `ろ`, for example.

So how to map this `ç` key onto ibus-mozc's Romaji Table?

(using ibus-mozc 2.20.2673.102)

**Comment by hiroyuki-komatsu** [comment]
Hi, please tell us which one you want to do.

1. type `ç` with Mozc.
2. convert `ç` to another character.

Thank you,

**Comment by paulolimac** [comment]
It is the second case.
I want to type my <kbd>ç</kbd> key and convert it to a japanese symbol.
For example: type <kbd>ç</kbd> and convert to `〜`.
(i think it is 2 bugs: one from `ç` and other from `~` chars).

I'm clicking on Mozc's `Properties` > `General` tab > Keymap section > `customize` button; to edit this romanji-hiragana table:
https://github.com/google/mozc/blob/master/src/data/preedit/romanji-hiragana.tsv

Using a old Mozc here, version (2.20.2673.102).
And I'm lack of practice about typing in japanese :smile:
But I think it is bugged in this feature (romanji-hiragana table editor) in many ways, because I got these strange cases:

1. When I edit the romanji-hiragana table, inputting a <kbd>ç</kbd> to convert to a japanese symbol, it only outputted/printed a pure `ç` (in other words: didn't convert it).
2. When I type <kbd>`~`</kbd>, to be converted in `〜`, it didn't convert again.
2.1. When using my default ibus's pt-br config, I need to type <kbd>`~`</kbd>+<kbd>space</kbd> to output/show the latin `~`.
Maybe, it is necessary to type a combo with <kbd>space</kbd>.
And I dont know if it is a issue from Ibus or Mozc.
2.2. I only got the `〜` when I edit/change the table input, from `~` to another key that doesn't need an +<kbd>space</kbd> combo. For example converting from `'` worked well.
3. When I edit the romanji-hiragana table, for example the first line from `-` to `---`, the last one converted properly to `ー`; but when I undo this change on the table, i.e. re-input the `-` to convert to `ー` it doesn't convert, even if I click on `apply` changes button or `reset to defaults` button.
3.1. Sometimes, in a random way, it worked without problems.
4. Sometimes, after editing the romanji-hiragana table, I didn't get the changes. So I reopen the text editor and re-set the ibus to input by mozc; but the inputs didn't works from my romanji-hiragana table new edits.

Again, I'm using a old version of Ibus-mozc, and maybe I missed something.

**Comment by hiroyuki-komatsu** [comment]
When you type <kbd>space</kbd>, Mozc changes the state from "alphabet->hiragana conversion (e.g. a → あ)" to "hiragana->kanji conversion (e.g. あ → 亜)". The rule of romanji-hiragana.tsv only works for "alphabet-hiragana conversion".

If your keyboard setting enables you to type `ç` as well as other alphabet characters, the logic of "alphabet->hiragana conversion" should work with `ç`. 

If screenshots or TSV of your setting is available, I'd be able to say something more.
Thanks,
